tesseract/Dockerfile

```diff
@@ -0,0 +1,90 @@
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    ros-$ROSDISTRO-ompl \
+    python-catkin-tools \
+    ros-$ROSDISTRO-octomap-msgs \
+    ros-$ROSDISTRO-octomap \
+    libassimp-dev \
+    libtinyxml-dev \
+    tar \
+    wget \
+    ros-$ROSDISTRO-rviz \
+    ros-$ROSDISTRO-octomap-ros \
+    clang \
+    clang-tidy \
+    libomp-dev \
+    lcov \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN git clone --branch master https://github.com/bulletphysics/bullet3.git
+RUN mkdir /bullet3_build && \
+    cd /bullet3_build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w -std=c++14" -DCMAKE_C_FLAGS="-w" -DBUILD_UNIT_TESTS=OFF -DBUILD_SHARED_LIBS=ON -DBULLET2_MULTITHREADING=OFF -DUSE_DOUBLE_PRECISION=ON ../bullet3 && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm -rf /bullet3 /bullet3_build 
+    
+RUN git clone --branch master https://github.com/danfis/libccd.git
+RUN mkdir /libccd_build && \
+    cd /libccd_build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w -std=c++14" -DCMAKE_C_FLAGS="-w" -DENABLE_DOUBLE_PRECISION=ON ../libccd && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm -rf /libccd /libccd_build 
+
+ENV PKG_CONFIG_PATH /opt/ros/$ROSDISTRO/lib/pkgconfig
+ENV LD_LIBRARY_PATH /opt/ros/$ROSDISTRO/lib
+RUN git clone --branch master https://github.com/flexible-collision-library/fcl.git
+RUN mkdir /fcl_build && \
+    cd /fcl_build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w -std=c++14" -DBUILD_TESTING=OFF ../fcl && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm -rf /fcl /fcl_build 
+
+RUN wget http://prdownloads.sourceforge.net/swig/swig-4.0.1.tar.gz
+RUN tar -xvf swig-4.0.1.tar.gz
+RUN cd /swig-4.0.1 && \
+    ./configure && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm swig-4.0.1.tar.gz && \
+    rm -rf /swig-4.0.1 && \
+    ln -s /usr/local/bin/swig /usr/local/bin/swig4.0 && \
+    ln -s /usr/local/bin/ccache-swig /usr/local/bin/ccache-swig4.0
+
+RUN git clone --branch release-1.8.1 https://github.com/google/googletest.git
+RUN mkdir /gtest_build && \
+    cd /gtest_build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w -std=c++14" -DBUILD_GMOCK=OFF -DBUILD_GTEST=ON -DBUILD_SHARED_LIBS=ON ../googletest && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm -rf /gtest /gtest_build 
+
+RUN if [ "$ROSDISTRO" = "kinetic" ] ; then git clone --branch v1.5.0 https://github.com/google/benchmark.git ; fi
+RUN if [ "$ROSDISTRO" = "kinetic" ] ; then \
+    mkdir /gbenchmark_build && \
+    cd /gbenchmark_build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w -std=c++14" -DBENCHMARK_ENABLE_GTEST_TESTS=OFF -DBUILD_SHARED_LIBS=ON ../benchmark && \
+    make -j8 && \
+    make install && \
+    cd .. && \
+    rm -rf /benchmark /gbenchmark_build ; fi
+
+ENV LD_LIBRARY_PATH /usr/local/lib:$LD_LIBRARY_PATH
```

Comment:
Is this really necessary?

Are dependent targets not linking to libs with absolute paths?

IIRC, `/usr/local/lib` should already be on the standard linker search path.

Reply:
I don't believe it is a default path on Ubuntu based on [this](https://askubuntu.com/questions/165027/why-cant-my-program-find-the-libraries-in-usr-local-lib) discussion.

Reply:
On both Ubuntu 16.04 and Ubuntu 18.04, the following is true for me:

```shell
$ cat /etc/ld.so.conf.d/libc.conf
# libc default configuration
/usr/local/lib
$ dpkg -S /etc/ld.so.conf.d/libc.conf
libc-bin: /etc/ld.so.conf.d/libc.conf
```

The Docker image used here also seems to contain the same file:

```
$ docker run -it --rm ros:melodic-perception cat /etc/ld.so.conf.d/libc.conf
# libc default configuration
/usr/local/lib
```

In any case: did things not work without it?

In general `LD_LIBRARY_PATH` is not something which should be set unless really needed.

Reply:
I don't believe it was, which is why I think it was added. I can remove it for now and if it does not work I will create a separate PR adding it back.

Reply:
If it is needed, it would be an indication of something else not being correct.